src/server/index.ts

```diff
@@ -131,16 +131,80 @@ export class DocumentDriveServer extends BaseDocumentDriveServer {
         this.initializePromise = this._initialize();
     }
 
-    private updateSyncStatus(
-        driveId: string,
-        status: SyncStatus | null,
+    private getOperationSource(source: StrandUpdateSource) {
+        return source.type === 'local' ? 'push' : 'pull';
+    }
+
+    private getCombinedSyncUnitStatus(
+        syncUnitStatus: SyncUnitStatusObject
+    ): SyncUnitStatus {
+        if (syncUnitStatus.pull === 'INITIAL_SYNC') return 'INITIAL_SYNC';
```

Comment:
if no pull has been executed yet, the status is going to be always 'INITIAL_SYNC'